app/models/requests/DataRequest.scala

```diff
@@ -26,6 +26,7 @@ case class OptionalDataRequest[A] (
 ) extends WrappedRequest[A](request) {
 
   def pptReference: String = request.pptReference
+  def cacheKey: String     = s"$userId-$pptReference"
```

Comment:
is it worth removing the userId parameter from these classes so no one is tempted to use it?

Reply:
this can just be: 
```suggestion
  def cacheKey: String     = request.cacheKey
```
ensures only constructed in one place